### ISSUE TITLE
Support for `cargo install --git <url>`

### ISSUE THREAD

**Issue by EtaoinWu** [baptiste0928/cargo-install#13]
Hi, I'd like to install Rust projects that are on github but not on crate.io.

To be specific, projects like `iden3/circom` [on github](https://github.com/iden3/circom) are [not uploaded to crate.io](https://github.com/iden3/circom/issues/40). I can install them with `cargo install --git <url>` without caching, and it's possible to fetch the github API to check the repo's latest tag/release for caching.

**Comment by baptiste0928** [comment]
Hi, thanks for the suggestion. This is definitely a use case that could be implemented.

`cargo` supports installing from git with a specific branch, commit or tag. The last two can be used directly for the cache key (although tags can be moved to another commit, this is rarely the case). For branches, I can easily get the commit hash from the [GitHub API](https://docs.github.com/en/rest/branches/branches#list-branches), but I can also try using [`git ls-remote`](https://git-scm.com/docs/git-ls-remote) to be compatible with *any* git repository.

For the default behavior without a specific commit/tag/branch, I hesitate between using the latest commit from the default branch or a "smarter" strategy (latest release, or latest tag if unavailable). But the latter doesn't follow the `cargo install` behavior I'm trying to mimic, since this action should be a drop-in replacement for uncached `cargo install` commands.

What do you think?

**Comment by baptiste0928** [comment]
I've implemented this in #14. Can you test if everything works for you before I merge it?

You can use the PR's branch like this:
```yaml
-  uses: baptiste0928/cargo-install@feat/git-install
   with:
    crate: circom  # required even for git installation (used in cache key and logs)
    git: https://github.com/iden3/circom
    tag: v2.1.5 # optional, rev/branch also supported
```

Let me know if it works or if run into any issue.

**Comment by EtaoinWu** [comment]
I have tested with your code, and it works. I have tested it without and with cache, and I tried specifying tag and rev.

I suggest that `rev` can be renamed to `commit`.

**Comment by EtaoinWu** [comment]
> I suggest that `rev` can be renamed to `commit`.

Disregard this suggestion -- I just learned that they are [different](https://stackoverflow.com/questions/11792538/in-git-what-is-the-difference-between-a-commits-and-a-revisions).

**Comment by baptiste0928** [comment]
> I have tested with your code, and it works. I have tested it without and with cache, and I tried specifying tag and rev.

Thanks for testing, I'll release a new version with these changes tomorrow or this weekend.

> I suggest that `rev` can be renamed to `commit`.
> Disregard this suggestion -- I just learned that they are [different](https://stackoverflow.com/questions/11792538/in-git-what-is-the-difference-between-a-commits-and-a-revisions).

I didn't know the difference either (I was following the names used by `cargo`). My code actually assumes that the revision refers to a unique commit, so `HEAD@{5 minutes ago}` will cause caching problems (the cache will be reused even if the last commit has changed). Even simple revisions like `HEAD` will not work.

This could be solved by using [`git rev-parse`](https://git-scm.com/docs/git-rev-parse), but this would require cloning the repository locally. I currently use [`git ls-remote`](https://git-scm.com/docs/git-ls-remote.html) to get commits corresponding to branches/tags without cloning.

So we can either :
- Clone the repository to resolve revisions, which will slow down the action for both cached and uncached runs (since it needs to be done to resolve the cache key).
- Keep the current behavior and rename `rev` to `commit` to avoid confusion.

Since using the more advanced revision syntax seems to be a rare use case, I think we can stick with the current version. This can always be changed in the future if someone asks.

**Comment by baptiste0928** [comment]
[`cargo install` seems to interpret `--rev` as a commit](https://doc.rust-lang.org/cargo/commands/cargo-install.html#install-options) and not an actual revision:

> `--rev` *sha*
>  Specific commit to use when installing from git.

So we should stick with `rev` since my code has the same behavior as cargo.

**Comment by EtaoinWu** [comment]
> So we should stick with `rev` since my code has the same behavior as cargo.

I agree with the consistency with cargo. Can you make `commit` as an alias of `rev`?

**Comment by baptiste0928** [comment]
I just published git installation support with [v2.1.0](https://github.com/baptiste0928/cargo-install/releases/tag/v2.1.0).
